projects/bp-gallery/src/components/views/search/SearchView.tsx

```diff
@@ -59,11 +61,32 @@ const SearchView = () => {
       searchTerms: allSearchTerms.filter(searchTerm => !isValidTimeSpecification(searchTerm)),
       searchTimes,
     };
-    // }
-    // return convertSearchParamsToPictureFilters(searchParams);
-  }, [/*isAllSearchActive,*/ searchParams]);
-  if (import.meta.env.MODE === 'development')
-    getSearchResultPictureIds(queryParams, '').then(res => console.log('search results:', res));
+  }, [searchParams]);
+
+  const [searchResultIds, error, state] = usePromise(
+    async () =>
+      (await getSearchResultPictureIds(queryParams, '')).map(hit => (hit.id as number).toString()),
+    [queryParams]
+  );
+
+  const pictureFilter: PictureFiltersInput = useMemo(() => {
+    if (error) {
+      console.log(error);
+      return {};
+    }
+    return state === 'resolved' ? { id: { in: searchResultIds } } : {};
+  }, [searchResultIds, state, error]);
+  console.log(pictureFilter);
+  const isOldSearchActive = useFlag('old_search');
+
+  if (import.meta.env.MODE === 'development') {
+    // getSearchResultPictureIds(queryParams, '').then(res => console.log('search results:', res));
+    console.log('resultids', searchResultIds);
+    console.log(
```

Comment:
do you still need all of these console logs?

Reply:
No, I don't. I will remove them once the pr is ready to be merged